### PR TITLE
credentials - aws - drop eksctl utility usage in onboarding script

### PR DESCRIPTION
**Pull request by ernestas2k**

### CODE REVIEW

File: docs/getting-started/credentials/configuring-eks-credentials/script.sh [hunk 1]
```diff
@@ -71,9 +69,13 @@ done
 aws iam put-user-policy --user-name $USER_NAME --policy-name CastEKSRestrictedAccess --policy-document $INLINE_POLICY_JSON
 
 echo "Adding user to cluster '$CLUSTER_NAME'"
-if ! eksctl get iamidentitymapping --cluster $CLUSTER_NAME --region $REGION --arn $USER_ARN -v 0 >>/dev/null 2>&1; then
-  eksctl create iamidentitymapping --cluster $CLUSTER_NAME --region $REGION --arn ${USER_ARN} --group system:masters --username $USER_NAME -C false -v 2
+CLUSTER_USER="- groups:\n  - system:masters\n  userarn: ${USER_ARN}\n  username: ${USER_NAME}\n"
+AWS_AUTH_CM_FILE=aws-auth-cm.json
+kubectl get -n=kube-system cm/aws-auth -o json >$AWS_AUTH_CM_FILE
+if [ "$(jq ".data.mapUsers | contains(\"${CLUSTER_USER}\")" <$AWS_AUTH_CM_FILE)" = "false" ]; then
```
Comment (varnastadeus) [inline docs/getting-started/credentials/configuring-eks-credentials/script.sh:75]: drop `jq`, you can patch without jq pretty easily

File: docs/getting-started/credentials/configuring-eks-credentials/script.sh [hunk 2]
```diff
@@ -18,26 +18,24 @@ if ! [ -x "$(command -v aws)" ]; then
...
...,\"Condition\":{\"StringEquals\":{\"ec2:ResourceTag/kubernetes.io/cluster/${CLUSTER_NAME}\":[\"owned\",\"shared\"]}}},{\"Sid\":\"VpcRestrictedActions\",\"Effect\":\"Allow\",\"Action\":[\"ec2:RevokeSecurityGroupIngress\",\"ec2:RevokeSecurityGroupEgress\",\"ec2:AuthorizeSecurityGroupEgress\",\"ec2:AuthorizeSecurityGroupIngress\",\"ec2:DeleteSecurityGroup\"],\"Resource\":\"*\",\"Condition\":{\"StringEquals\":{\"ec2:Vpc\":\"arn:aws:ec2:${ARN}:vpc/${VPC}\"}}},{\"Sid\":\"AutoscalingActionsTagRestriction\",\"Effect\":\"Allow\",\"Action\":[\"autoscaling:UpdateAutoScalingGroup\",\"autoscaling:DeleteAutoScalingGroup\",\"autoscaling:SuspendProcesses\",\"autoscaling:ResumeProcesses\",\"autoscaling:TerminateInstanceInAutoScalingGroup\"],\"Resource\":\"arn:aws:autoscaling:${ARN}:autoScalingGroup:*:autoScalingGroupName/*\",\"Condition\":{\"StringEquals\":{\"autoscaling:ResourceTag/kubernetes.io/cluster/${CLUSTER_NAME}\":[\"owned\",\"shared\"]}}},{\"Sid\":\"EKS\",\"Effect\":\"Allow\",\"Action\":[\"eks:Describe*\",\"eks:List*\",\"eks:DeleteNodegroup\",\"eks:UpdateNodegroupConfig\"],\"Resource\":[\"arn:aws:eks:${ARN}:cluster/${CLUSTER_NAME}\",\"arn:aws:eks:${ARN}:nodegroup/${CLUSTER_NAME}/*/*\"]}]}"
```
Comment (varnastadeus) [inline docs/getting-started/credentials/configuring-eks-credentials/script.sh:40]: you forgot to update json

File: docs/getting-started/credentials/configuring-eks-credentials/script.sh [hunk 3]
```diff
@@ -18,26 +18,24 @@ if ! [ -x "$(command -v aws)" ]; then
...
...tion\":{\"StringEquals\":{\"ec2:ResourceTag/kubernetes.io/cluster/${CLUSTER_NAME}\":[\"owned\",\"shared\"]}}},{\"Sid\":\"VpcRestrictedActions\",\"Effect\":\"Allow\",\"Action\":[\"ec2:RevokeSecurityGroupIngress\",\"ec2:RevokeSecurityGroupEgress\",\"ec2:AuthorizeSecurityGroupEgress\",\"ec2:AuthorizeSecurityGroupIngress\",\"ec2:DeleteSecurityGroup\"],\"Resource\":\"*\",\"Condition\":{\"StringEquals\":{\"ec2:Vpc\":\"arn:aws:ec2:${ARN}:vpc/${CLUSTER_VPC}\"}}},{\"Sid\":\"AutoscalingActionsTagRestriction\",\"Effect\":\"Allow\",\"Action\":[\"autoscaling:UpdateAutoScalingGroup\",\"autoscaling:DeleteAutoScalingGroup\",\"autoscaling:SuspendProcesses\",\"autoscaling:ResumeProcesses\",\"autoscaling:TerminateInstanceInAutoScalingGroup\"],\"Resource\":\"arn:aws:autoscaling:${ARN}:autoScalingGroup:*:autoScalingGroupName/*\",\"Condition\":{\"StringEquals\":{\"autoscaling:ResourceTag/kubernetes.io/cluster/${CLUSTER_NAME}\":[\"owned\",\"shared\"]}}},{\"Sid\":\"EKS\",\"Effect\":\"Allow\",\"Action\":[\"eks:Describe*\",\"eks:List*\",\"eks:DeleteNodegroup\",\"eks:UpdateNodegroupConfig\"],\"Resource\":[\"arn:aws:eks:${ARN}:cluster/${CLUSTER_NAME}\",\"arn:aws:eks:${ARN}:nodegroup/${CLUSTER_NAME}/*/*\"]}]}"
```
Comment (varnastadeus) [inline docs/getting-started/credentials/configuring-eks-credentials/script.sh:38]: you forgot to update json

File: docs/getting-started/credentials/configuring-eks-credentials/script.sh [hunk 4]
```diff
@@ -18,26 +18,24 @@ if ! [ -x "$(command -v aws)" ]; then
...
 fi
 
-if ! [ -x "$(command -v eksctl)" ]; then
-  echo "Error: eksctl is not installed"
-  exit 1
-fi
-
 echo "Fetching cluster information"
-eksctl get cluster --region $REGION $CLUSTER_NAME -C false -v 2
+CLUSTER=$(aws eks describe-cluster --name $CLUSTER_NAME --region $REGION --output json)
```
Comment (anjmao) [inline docs/getting-started/credentials/configuring-eks-credentials/script.sh:22]: Maybe check that `jq` exist
```sh
if ! [ -x "$(command -v jq)" ]; then
  echo "Error: jq is not installed"
  exit 1
fi
```

Comment (ernestas2k) [reply docs/getting-started/credentials/configuring-eks-credentials/script.sh:22]: Good idea 👍

**Review by anjmao — APPROVED** [review]

File: docs/getting-started/credentials/configuring-eks-credentials/script.sh (hunk 1 above)
Comment (ernestas2k) [reply docs/getting-started/credentials/configuring-eks-credentials/script.sh]: Let me create a ticket to address jq usage, we would need to throw out jq usage in curl POST requests to cast api (already discussed approach) and change the way we edit configmap in this exact place.

**Pull request merged**